### ISSUE TITLE
Convert draw/_draw.pyx to "r/c" naming conventions

### ISSUE THREAD

**Issue by JDWarner** [scikit-image/scikit-image#1602]
Currently the Cython file `_draw.pyx` uses "x/y" naming conventions in docstrings and variable naming. Apply a style fix, simply renaming variables as appropriate.

**Comment by soupault** [comment]
Isn't `m/n` the convention for image processing?

**Comment by JDWarner** [comment]
@soupault we standardized on r/c because it actually carries meaning, in the form of row/column. x/y, m/n are all dangerous and potentially unintuitive because 2d Numpy arrays have the origin in the upper left and, when printed, the first dimension is rows while the second is columns.

After a lot of support requests boiling down to "why is indexing not working like I expect it to" we settled on r/c.

**Comment by soupault** [comment]
@JDWarner I see. So `r/c` actually matches traditional `numpy`-notation?

**Comment by JDWarner** [comment]
Not just numpy; Matlab also orients matrices this way.

**Comment by soupault** [comment]
As we exposed a lot of functions in https://github.com/scikit-image/scikit-image/issues/2303, I argue that this issue must be a blocker for 0.13 release (so to avoid unnecessary API change deprecations).

**Comment by jni** [comment]
@soupault I agree with your argument. =)

**Comment by pamitabh** [comment]
I was checking the draw module and see that everything except the draw.ellipsoid follows this "r/c" naming convention. Is that something that needs fixing? Should I open a new issue for that?

**Comment by jni** [comment]
Hey @pamitabh,

Ellipsoid doesn't return coordinates, only a binary array, which is why we made `ellipsoid_coords`. In short, I don't think this issue applies to ellipsoid. We may think about a future where all draw functions return coordinates, but for now, there is nothing to do about ellipsoid other than think about deprecating it.

**Comment by jni** [comment]
... Sorry, I just saw your post in #7239. 🤦‍♂️ I agree that the nomenclature is confusing. The question is whether the *ordering* of the axes needs changing, or only the documentation. Based on some quick testing:

```python
In [1]: draw.ellipsoid(100, 10, 1).shape
Out[1]: (203, 23, 5)

In [2]: draw.ellipsoid(10, 1, 100).shape
Out[2]: (23, 5, 203)
```

it's just the documentation that needs to change — replacing x/y/z with p/r/c.

**Comment by pamitabh** [comment]
I agree about x/y/z <-> p/r/c. I have more thoughts about this, but it concerns with defining rotation angles. More at #7239
